main.go

```diff
@@ -169,6 +179,61 @@ func main() {
 	os.Exit(evaluateOutcome(summary, test.Config.Expected)) // Returns success on all tests to OS; this allows for test scripting.
 }
 
+/* parse all input args of the form `parameter=value`, e.g. `X=5` */
+func parseArgs(args []string) map[string]string {
+	parsedArgs := make(map[string]string)
+	for _, arg := range args {
+		splitArgs := strings.Split(arg, "=")
+		if len(splitArgs) != 2 {
+			fmt.Println("Bad assignment: " + arg)
+			os.Exit(1)
+		}
+
+		param, val := splitArgs[0], splitArgs[1]
+		parsedArgs[param] = val
+	}
+	return parsedArgs
+}
+
+/* resolve all parameters in test file */
+func replaceAllParams(fileData []byte, params map[string]string) []byte {
+	// replace all given parameters with their specified values
+	fileData = replaceInputParams(fileData, params)
+	// replace all other parameters with their default values
+	fileData = replaceDefaultParams(fileData)
+
+	return fileData
+}
+
+func compileParamRegex(param string) *regexp.Regexp {
+	// matches strings of the form `%{parameter}` and `%{parameter, value}`
+	paramRegex, _ := regexp.Compile("%\\{\\s*" + param + "\\s*(?:,\\s*.*?\\s*)?\\}")
```

Comment:
@hsanjuan Gotcha, so it sounds like we'd want to do something like:

1. Load the parameters-config file if passed via CLI flag, otherwise load `./vars.yml` if it exists.
2. Parse parameter replacements passed as CLI flags, override any that were found in the parameters-config.

How does that sound/is there anything I'm missing there?